### PR TITLE
[Integration][Airflow] Do not change task instance during on_running event.

### PR DESCRIPTION
**Pull request by JDarDagran**
Signed-off-by: Jakub Dardzinski <kuba0221@gmail.com>

### Problem

`on_running` hook changed `TaskInstance` object along with `task` attribute. The issue turned out when dealing with Pandas DataFrames as resolved template fields.

Closes: #1017 

### Solution

Work on copy of task instance in OL hook.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [ ] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

### Checklist

- [x] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [x] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [x] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] You've updated the [`CHANGELOG.md`](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md) with details about your change under the "Unreleased" section (_if relevant, depending on the change, this may not be necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [x] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

### CODE REVIEW

File: integration/airflow/openlineage/airflow/listener.py [hunk 1]
```diff
@@ -87,11 +88,12 @@ def on_task_instance_running(previous_state, task_instance: "TaskInstance", sess
     dag = task_instance.task.dag
 
     def on_running():
-        task_instance.render_templates()
-        task = task_instance.task
+        new_task_instance = copy.deepcopy(task_instance)
```
Comment (pawel-big-lebowski) [inline integration/airflow/openlineage/airflow/listener.py:91]: if it is a deepcopy, we can create a variable name with `deepcopy` or `copy` suffix.

File: integration/airflow/tests/test_listener.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    )
+    run_id = str(uuid.uuid1())
+    dag.create_dagrun(state=State.NONE, run_id=run_id)
+    ti = TaskInstance(t, run_id=run_id)
+    ti.check_and_change_state_before_execution()  # make listener hook on running event
+    ti._run_raw_task()
+    # we need to unregister hooks not to break BigQuery E2E tests
+    unregister_task_instance_state_events()
+    pd.testing.assert_frame_equal(xcom_push_mock.call_args[1]["value"], render_df())
+    assert not isinstance(render_mock.call_args[0][0], pd.DataFrame)
```
Comment (pawel-big-lebowski) [inline integration/airflow/tests/test_listener.py:52]: Please do add some some inline comments to the test, especially for the two assertions.

**Review by pawel-big-lebowski — COMMENTED** [review]
This seems to be a valid solution to a problem. 
Some minor comments added.

File: integration/airflow/tests/test_listener.py (hunk 2 above)
Comment (JDarDagran) [reply integration/airflow/tests/test_listener.py:56]: 👍

**Review by pawel-big-lebowski — APPROVED** [review]

**Pull request merged**